Lib/Src/CAdcChannel.cpp

```diff
@@ -0,0 +1,83 @@
+/**
+ *@file CAdcChannel.cpp
+ *@brief This files contains the methods for the CAdcChannel class
+ *
+ */
+
+/*
+ *  Created on: Feb 14, 2022
+ *      Author: niuslar
+ */
+
+#include "CAdcChannel.h"
+
+#define ADC_CONVERSION_COEFF ((float)ADC_VDDA / ADC_RES)
+
+bool CAdcChannel::s_adc_initialised = false;
+ADC_HandleTypeDef *CAdcChannel::sp_hadc = nullptr;
+uint16_t CAdcChannel::s_adc_data_buf[ADC_CHANNELS];
+
+/**
+ * @brief Constructor
+ * @param adc channel
+ */
+CAdcChannel::CAdcChannel(uint8_t adc_channel) : m_adc_channel(adc_channel)
+{
+    if (adc_channel >= ADC_CHANNELS)
+    {
+        Error_Handler();
+    }
+}
+
+/**
+ * @brief Calibrate and Start ADC
+ * @note The ADC will read all the channels once and then stop.
+ * To read the channels again, use the triggerADC() method.
+ * @note This function needs to be called only once.
+ */
+void CAdcChannel::startADC(ADC_HandleTypeDef *p_hadc)
```

Comment:
This is called startADC even when it's part of the CAdcChannel class because it starts the general ADC and needs to be called only once.